### PR TITLE
[13.0][MIG] website_snippet_carousel_product: Migration to v13

### PR DESCRIPTION
**Pull request by CarlosRoca13**
@Tecnativa TT26960

This Pr is WIP because the interval function of the file `snippet.option.js` is giving a random error at edit mode that I can't solve. Can someone give me some help?

Please @Tardo @pedrobaeza review this

### CODE REVIEW

File: website_snippet_carousel_product/static/src/js/s_product_carousel_frontend.js [hunk 1]
```diff
@@ -3,24 +3,25 @@
...
-    var sAnimation = require("website.content.snippets.animation");
+    const core = require("web.core");
+    const sAnimation = require("website.content.snippets.animation");
 
-    var _t = core._t;
+    const _t = core._t;
 
     sAnimation.registry.js_product_carousel = sAnimation.Class.extend({
         selector: ".js_product_carousel",
+        disabledInEditableMode: false,
```
Comment (chienandalu) [inline website_snippet_carousel_product/static/src/js/s_product_carousel_frontend.js:13]: Why?

Comment (CarlosRoca13) [reply website_snippet_carousel_product/static/src/js/s_product_carousel_frontend.js:13]: It is needed for see the preview of the snippet... If we don't use this flag it is initialized as true and we couldn't see the preview.

Comment (Tardo) [reply website_snippet_carousel_product/static/src/js/s_product_carousel_frontend.js:13]: More info here: https://github.com/odoo/odoo/blob/76c9ba9d87ec7069d3c11e605bb85b01091e2b32/addons/website/static/src/js/content/snippets.animation.js#L49

File: website_snippet_carousel_product/static/src/js/s_product_carousel_frontend.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+                    }
+
+                    _this.$target.html($object_html);
+                    _this.$target.find(".carousel").carousel({
+                        interval: interval,
+                    });
+                    // Initialize 'animations' for the product card.
+                    // This is necessary because the snippet is asynchonously
+                    // rendered on the server.
+                    _this.trigger_up("animation_start_demand", {
```
Comment (Tardo) [inline website_snippet_carousel_product/static/src/js/s_product_carousel_frontend.js]: This is necessary for the buttons to work
```suggestion
                    _this.trigger_up("widgets_start_request", {
```

File: website_snippet_carousel_product/static/src/js/s_product_carousel_frontend.js [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+            const domain = this.$target.attr("data-domain") || "[]";
+            const products_per_slide =
+                Number(this.$target.attr("data-products-per-slide")) || 4;
+            let interval = Number(this.$target.attr("data-interval"));
+            if (_.isNaN(interval)) {
+                interval = 5000;
+            }
+
+            // Prevent user edition
+            this.$target.attr("contenteditable", "False");
```
Comment (Tardo) [inline website_snippet_carousel_product/static/src/js/s_product_carousel_frontend.js]: The values on the documentation are in lowercase...
```suggestion
            this.$target.attr("contenteditable", "false");
```
But maybe you can use "o_not_editable" css class directly in the snippet element.

File: website_snippet_carousel_product/static/src/js/snippet.options.js [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+            return wUtils
+                .prompt({
+                    id: this.popup_template_id,
+                    window_title: this.popup_title,
+                    input: _t("Domain (can be empty)"),
+                    init: function() {
+                        return _this.$target.attr("data-domain");
+                    },
+                })
+                .then(domain => this.set_domain(domain))
```
Comment (Tardo) [inline website_snippet_carousel_product/static/src/js/snippet.options.js]:
```suggestion
                .then(domain => this.set_domain(domain.val))
```

File: website_snippet_carousel_product/static/src/js/snippet.options.js [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+                .catch(() => this.set_domain());
+        },
+
+        /**
+         * @param {String} domain
+         */
+        set_domain: function(domain) {
+            const sdomain = domain || "";
+            this.$target.attr("data-domain", sdomain.val.replace(/'/g, '"'));
```
Comment (Tardo) [inline website_snippet_carousel_product/static/src/js/snippet.options.js]:
```suggestion
            this.$target.attr("data-domain", sdomain.replace(/'/g, '"'));
```

**Review by Tardo — CHANGES_REQUESTED** [review]
I can't reproduce the issue with "interval" operation... maybe now must use "Speed" option of the carousel snippet instead of add a custom option in the product carousel snippet.

File: website_snippet_carousel_product/static/src/js/s_product_carousel_frontend.js (hunk 1 above)
Comment (chienandalu) [reply website_snippet_carousel_product/static/src/js/s_product_carousel_frontend.js:13]: Thanks, @Tardo :+1:

**Comment by CarlosRoca13** [comment]
I could not make the Speed option work... But no one of the carousel options added at that Block are workig. So I achieve to not show that options. Can u review it again @Tardo??

**Review by Tardo — APPROVED** [review]
LGTM :+1:

**Review by chienandalu — APPROVED** [review]

**Pull request merged**